sonaranalyzer-dotnet/build/build-utils.ps1

```diff
@@ -0,0 +1,330 @@
+Add-Type -AssemblyName "System.IO.Compression.FileSystem"
+
+$sonarqube_runner_version = "2.0"
+
+# Resolves the given relative to the repository path to absolute.
+function Resolve-RepoPath([string]$relativePath) {
+    return (Join-Path (Resolve-Path (Join-Path $PSScriptRoot "..")) $relativePath)
+}
+
+# Original: http://jameskovacs.com/2010/02/25/the-exec-problem
+function Exec ([scriptblock]$command, [string]$errorMessage = "Error executing command: " + $command) {
+    $output = & $command
+    if ((-not $?) -or ($lastexitcode -ne 0)) {
+        Write-Error $output
+        throw $errorMessage 
+    }
+    return $output
+}
+
+function Test-ExitCode([string]$errorMessage = "Error executing command.") {
+    if ((-not $?) -or ($lastexitcode -ne 0)) {
+        throw $errorMessage 
+    }
+}
+
+# Sets the current folder and executes the given script.
+# When the script finishes sets the original current folder.
+function Exec-InLocation([string]$path, [scriptblock]$command) {
+    try {
+        Push-Location $path
+        & $command
+    }
+    finally {
+        Pop-Location
+    }
+}
+
+function To-UnixLineEndings([string]$fileName) {
+    Get-ChildItem $fileName | ForEach-Object {
+        $content = [IO.File]::ReadAllText($_) -Replace "`r`n?", "`n"
+        $utf8 = New-Object System.Text.UTF8Encoding $false
+        [IO.File]::WriteAllText($_, $content, $utf8)
+    }
+}
+
+function Write-Header([string]$text) {
+    Write-Host "================================================"
+    Write-Host $text
+    Write-Host "================================================"
+}
+
+## Build ############################################################
+
+function Get-BuildNumber([string]$default = "0") {
+    if ($env:BUILD_NUMBER) {
+        return $env:BUILD_NUMBER
+    }
+    return $default
+}
+
+function Get-BranchName {
+    if ($env:GITHUB_BRANCH) {
+        if ($env:GITHUB_BRANCH.StartsWith("refs/heads/")) {
+            return $env:GITHUB_BRANCH.Substring(11)
+        }
+        return $env:GITHUB_BRANCH
+    }
+    return Exec { & git rev-parse --abbrev-ref HEAD }
+}
+
+function Get-Sha1 {
+    if ($env:GIT_SHA1) {
+        return $env:GIT_SHA1
+    }
+    return Exec { & git rev-parse HEAD }
+}
+
+function Get-ExecutablePath([string]$name, [string]$envVar) {
+    $path = [environment]::GetEnvironmentVariable($envVar, "Process")
+
+    if (!$path) {
+        $path = Exec { & where.exe $name } `
+            | Select-Object -First 1
+    }
+
+    if (Test-Path $path) {
+        return $path

```

Comment:
Maybe we could put the Write-Host "Found '$name' at '$path'" here and not in calling methods.